communication/mavlink_waypoint_handler.cpp

```diff
@@ -517,1514 +384,288 @@ void Mavlink_waypoint_handler::clear_waypoint_list(Mavlink_waypoint_handler* way
+    callback.module_struct  = (Mavlink_message_handler::handling_module_struct_t) this;
+    init_success &= message_handler_.add_msg_callback(&callback);
 
-        result = MAV_RESULT_ACCEPTED;
-    }
-    else
-    {
-        result = MAV_RESULT_TEMPORARILY_REJECTED;
+    callback.message_id     = MAVLINK_MSG_ID_MISSION_CLEAR_ALL; // 45
+    callback.sysid_filter   = MAVLINK_BASE_STATION_ID;
+    callback.compid_filter  = MAV_COMP_ID_ALL;
+    callback.function       = (Mavlink_message_handler::msg_callback_func_t)      &clear_waypoint_list;
+    callback.module_struct  = (Mavlink_message_handler::handling_module_struct_t) this;
 
-        print_util_dbg_print("Not ready to switch to next waypoint. Either no waypoint loaded or flying towards one\r\n");
-    }
+    callback.message_id     = MAVLINK_MSG_ID_MISSION_ACK; // 47
+    callback.sysid_filter   = MAVLINK_BASE_STATION_ID;
+    callback.compid_filter  = MAV_COMP_ID_ALL;
+    callback.function       = (Mavlink_message_handler::msg_callback_func_t)      &receive_ack_msg;
+    callback.module_struct  = (Mavlink_message_handler::handling_module_struct_t) this;
+    init_success &= message_handler_.add_msg_callback(&callback);
+
+    init_homing_waypoint();
 
-    // To avoid a MAV_RESULT_TEMPORARILY_REJECTED for the second message and thus
-    //  a bad information to the user on the ground, if two messages are received
-    //  in a short time interval, we still show the result as MAV_RESULT_ACCEPTED
-    if (time_from_start_wpt < time_wpt_limit)
+    if(!init_success)
     {
-        result = MAV_RESULT_ACCEPTED;
+        print_util_dbg_print("[MAVLINK_WAYPOINT_HANDLER] constructor: ERROR\r\n");
     }
 
-    return result;
+    return init_success;
 }
 
-mav_result_t Mavlink_waypoint_handler::is_arrived(Mavlink_waypoint_handler* waypoint_handler, mavlink_command_long_t* packet)
+void Mavlink_waypoint_handler::init_homing_waypoint()
 {
-    mav_result_t result;
+    /*
+    Constructor order:
+
+    uint8_t frame,
+    uint16_t command,
+    uint8_t autocontinue,
+    float param1,   Minimum pitch (if airspeed sensor present), desired pitch without sensor [rad]
+    float param2,   Empty
+    float param3,   Takeoff ascend rate [ms^-1]
+    float param4,   Yaw angle [rad] (if magnetometer or another yaw estimation source present), ignored without one of these
+    float param5,   Y-axis position [m] (I dont know why they made it this order)
+    float param6,   X-axis position [m] (I dont know why they made it this order)
+    float param7    Z-axis position [m]
+    */
+    Waypoint waypoint(  MAV_FRAME_LOCAL_NED,
+                        MAV_CMD_NAV_TAKEOFF,
+                        0,
+                        0.0f,
+                        0.0f,
+                        0.0f,
+                        0.0f,
+                        0.0f,
+                        0.0f,
+                        navigation_.takeoff_altitude);
```

Comment:
- can we get takeoff_altitude from elsewhere
- or move the corrdinates of the home waypoint to config